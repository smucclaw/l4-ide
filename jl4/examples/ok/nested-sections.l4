§ a
ASSUME a1 IS A BOOLEAN
§§ aa
ASSUME a2 IS A BOOLEAN
§§§ aaa
ASSUME a3 IS A BOOLEAN
#CHECK a2
#CHECK a3
#CHECK aaa.a3
-- this does not currently work: 
-- #CHECK a.aa.aaa.a3

§§ ab
ASSUME a4 IS A BOOLEAN
§§ ac
§ b
ASSUME a1 IS A BOOLEAN
§§ ba
ASSUME a2 IS A BOOLEAN
§ c

#CHECK b.a1
#CHECK b.a2
#CHECK b.ba.a2
#CHECK a.a1
#CHECK a.aa.a2
#CHECK a.aa.a3
#CHECK a.aa.aaa.a3
#CHECK a.ab.a4
#CHECK a.a4