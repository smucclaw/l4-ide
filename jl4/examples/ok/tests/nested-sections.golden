Parsing successful
Typechecking successful
nested-sections.l4:7:8-10:
  BOOLEAN
nested-sections.l4:8:8-10:
  BOOLEAN
nested-sections.l4:9:8-14:
  BOOLEAN
nested-sections.l4:23:8-12:
  BOOLEAN
nested-sections.l4:24:8-12:
  BOOLEAN
nested-sections.l4:25:8-15:
  BOOLEAN
nested-sections.l4:26:8-12:
  BOOLEAN
nested-sections.l4:27:8-15:
  BOOLEAN
nested-sections.l4:28:8-15:
  BOOLEAN
nested-sections.l4:29:8-19:
  BOOLEAN
nested-sections.l4:30:8-15:
  BOOLEAN
nested-sections.l4:31:8-16:
  BOOLEAN
nested-sections.l4:32:8-16:
  BOOLEAN
nested-sections.l4:33:8-16:
  NUMBER
